app/classifier/tasks/dropdown/index.cjsx

```diff
@@ -121,24 +120,30 @@ module.exports = React.createClass
   handleFocus: ->
     @menus[0].focus()
 
-  getOptions: (i, optionsKeys) ->
-    optionsKey = if optionsKeys.length then optionsKeys.join(';') else '*'
+  getOptionsKey: (i, optionsKey = '') ->
+    {selects} = @props.task
+    select = selects[i]
+    [condition] = selects.filter (filterSelect) => filterSelect.id is select.condition
```

Comment:
I think `clearRelated` kind of looks ahead and clears other menus that the menu that has changed controls, is that similar to what you're referring to?

I like the idea of the lookup table, would we store it in state? If so, that might be similar to the `optionsKeys` object Brian and I tried in the [old version](https://github.com/zooniverse/Panoptes-Front-End/commit/32c68b104b2f6d60c0327094696b23f8daafd2fb).